scripts/variantstore/wdl/GvsCreateVATAnnotations.wdl

```diff
@@ -0,0 +1,320 @@
+        ## normalize, left align and split multi allelic sites to new lines, remove duplicate lines
+        bcftools norm -m- --check-ref w -f Homo_sapiens_assembly38.fasta filtered.vcf.gz -Oz -o normalized.vcf.gz
+        rm ~{local_input_vcf}
+        ## filter out spanning deletions and variants with an AC of 0
+        bcftools view -e 'ALT[0]="*" || AC=0' --no-update normalized.vcf.gz -Ou | \
+        ## ensure that we respect the FT tag
+        bcftools filter -i "FORMAT/FT='PASS,.'" --set-GTs . -Oz -o ~{normalized_vcf}
+
+        ## clean up unneeded file
+        rm normalized.vcf.gz
+
+        ## During normalization, sometimes duplicate variamts appear but with different calculations. This seems to be a bug in bcftools. For now we are dropping all duplicate variants
+        ## The way in which this is done is a bit hamfisted and should be optimized in the future.
+        ## to locate the duplicates, we first make a file of just the first 5 columns
+        bcftools query ~{normalized_vcf} -f '%CHROM\t%POS\t%ID\t%REF\t%ALT\n' > check_duplicates.tsv
+        ## check it for duplicates and put them in a new file
+        sort check_duplicates.tsv | uniq -d | cut -f1,2,3,4,5  > duplicates.tsv
+        rm check_duplicates.tsv ## clean up
+        ## remove those rows (that match up to the first 5 cols)
+        zgrep -v -wFf duplicates.tsv ~{normalized_vcf} > deduplicated.vcf.gz
+        rm ~{normalized_vcf} ## clean up
+
+        ## add duplicates to the file that's tracking dropped variants
+        cat duplicates.tsv >> track_dropped.tsv
+        rm duplicates.tsv ## clean up unneeded file
+
+        ## calculate annotations for all subpopulations
+        ## AC_het,AC_hom and AC_Hemi are used to calculate the participant count
+        bcftools plugin fill-tags  -- deduplicated.vcf.gz -S ~{subpopulation_sample_list} -t AC,AF,AN,AC_het,AC_hom,AC_Hemi | bcftools query -f \
+        '%CHROM\t%POS\t%REF\t%ALT\t%AC\t%AN\t%AF\t%AC_Hom\t%AC_Het\t%AC_Hemi\t%AC_afr\t%AN_afr\t%AF_afr\t%AC_Hom_afr\t%AC_Het_afr\t%AC_Hemi_afr\t%AC_amr\t%AN_amr\t%AF_amr\t%AC_Hom_amr\t%AC_Het_amr\t%AC_Hemi_amr\t%AC_eas\t%AN_eas\t%AF_eas\t%AC_Hom_eas\t%AC_Het_eas\t%AC_Hemi_eas\t%AC_eur\t%AN_eur\t%AF_eur\t%AC_Hom_eur\t%AC_Het_eur\t%AC_Hemi_eur\t%AC_mid\t%AN_mid\t%AF_mid\t%AC_Hom_mid\t%AC_Het_mid\t%AC_Hemi_mid\t%AC_oth\t%AN_oth\t%AF_oth\t%AC_Hom_oth\t%AC_Het_oth\t%AC_Hemi_oth\t%AC_sas\t%AN_sas\t%AF_sas\t%AC_Hom_sas\t%AC_Het_sas\t%AC_Hemi_sas\n' \
+        >> ~{custom_annotations_file_name}
+
+        ## for validation of the pipeline
+        wc -l ~{custom_annotations_file_name} | awk '{print $1 -7}'  > count.txt
+
+        ## compress the vcf and index it, make it sites-only for the next step
+        bcftools view --no-update --drop-genotypes deduplicated.vcf.gz -Oz -o ~{normalized_vcf_compressed}
+        ## if we can spare the IO and want to pass a smaller file we can also drop the info field w bcftools annotate -x INFO
```

Comment:
not sure what this comment means

Reply:
it was in case we got to a point where the size of the files was an issue (which it sometimes was with the > 50 alt alleles) 
but for now, we're optimizing for the time / memory

Reply:
so, this is an additional step that can be run before the `bcftools index --tbi ~{normalized_vcf_compressed}` ?

Reply:
yeah---I dont want to forget it forever, but I'm not wedded to it